### PR TITLE
Geometry docstrings

### PR DESCRIPTION
**Pull request by tomvanmele**
### What type of change is this?

- [ ] Bug fix in a **backwards-compatible** manner.
- [ ] New feature in a **backwards-compatible** manner.
- [ ] Breaking change: bug fix or new feature that involve incompatible API changes.
- [ ] Other (e.g. doc update, configuration, etc)

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [ ] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [ ] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [ ] I added new functions/classes and made them available on a second-level import, e.g. `compas.datastructures.Mesh`.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by tomvanmele** [comment]
@gonzalocasas ready

File: NOTES [hunk 1]
```diff
@@ -0,0 +1,5 @@
+* polyline divide by length function does not make much sense (and returns an incorrect result)
```
Comment (gonzalocasas) [inline NOTES]: is this doc really supposed to be committed?

File: src/compas/geometry/_core/normals.py [hunk 2]
```diff
@@ -76,17 +78,19 @@ def normal_triangle(triangle, unitized=True):
...
     Returns
     -------
-    list
+    [float, float, float]
         The normal vector.
 
     Raises
     ------
-    ValueError
+    AssertionError
```
Comment (gonzalocasas) [inline src/compas/geometry/_core/normals.py:93]: this is something we might reconsider at some point: `assert` statements are removed and `AssertionError` are not thrown when python is run with optimizations enabled (`python -o`). I haven't checked this in a long while, but as far as I remember, `assert` should only be used in test cases, not in the rest of the source code.

In any case, that change -if we ever do it- is for a different PR.

File: src/compas/geometry/transformations/translation.py [hunk 3]
```diff
@@ -67,6 +73,14 @@ def __init__(self, matrix=None, check=True):
                     raise ValueError('This is not a proper translation matrix.')
         super(Translation, self).__init__(matrix=matrix)
 
+    @property
+    def translation_vector(self):
+        from compas.geometry import Vector
```
Comment (gonzalocasas) [inline src/compas/geometry/transformations/translation.py:78]: why the inlined import?

File: src/compas/geometry/primitives/frame.py [hunk 4]
```diff
@@ -645,29 +690,29 @@ def to_local_coordinates(self, object_in_wcf):
...
-        if isinstance(object_in_wcf, list):
-            return Point(*object_in_wcf).transformed(T)
-        else:
-            return object_in_wcf.transformed(T)
+        if isinstance(obj, (list, tuple)):
+            return Point(*obj).transformed(T)
+        return obj.transformed(T)
 
-    def to_world_coordinates(self, object_in_lcf):
+    def to_world_coordinates(self, obj):
```
Comment (gonzalocasas) [inline src/compas/geometry/primitives/frame.py]: i would tend to keep the `lcf` (or `wcf` ) references in param names, sometimes this thing is very confusing and it is not bad to have it explicit in the name.

File: src/compas/geometry/transformations/rotation.py [hunk 5]
```diff
@@ -69,25 +81,56 @@ def __init__(self, matrix=None, check=True):
                     raise ValueError('This is not a proper rotation matrix.')
         super(Rotation, self).__init__(matrix=matrix)
 
+    @property
+    def quaternion(self):
+        from compas.geometry import Quaternion
```
Comment (gonzalocasas) [inline src/compas/geometry/transformations/rotation.py:86]: same thing here.. why are these imports inlined?

**Review by gonzalocasas — APPROVED** [review]
LGTM + comments!

File: NOTES (hunk 1 above)
Comment (tomvanmele) [reply NOTES]: nope will remove

File: src/compas/geometry/transformations/translation.py (hunk 3 above)
Comment (tomvanmele) [reply src/compas/geometry/transformations/translation.py:78]: avoid circular import. the imports of transformations precede the imports of primitives etc.

File: src/compas/geometry/transformations/rotation.py (hunk 5 above)
Comment (tomvanmele) [reply src/compas/geometry/transformations/rotation.py:86]: same

Comment (tomvanmele) [reply src/compas/geometry/transformations/rotation.py:86]: this problem might go away after restructuring but that is for the next set of PRs :)

Comment (gonzalocasas) [reply src/compas/geometry/transformations/rotation.py:86]: 👍

**Pull request merged**